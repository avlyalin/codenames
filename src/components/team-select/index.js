export { TeamSelect } from './team-select';
