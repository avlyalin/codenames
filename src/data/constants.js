export const LANGUAGES = {
  ru: 'ru',
};

export const FIELD_SIZES = {
  '5x4': '5x4',
  '5x5': '5x5',
  '5x6': '5x6',
};

export const CARDS_DICTIONARIES = {
  GAGA: 'gaga',
};

export const CARDS_TYPES = {
  agent: 'agent',
  double_agent: 'double_agent',
  killer: 'killer',
  citizen: 'citizen',
};

export const TEAMS = {
  blue: 'blue',
  red: 'red',
};
