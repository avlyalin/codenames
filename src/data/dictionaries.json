{
  "gaga": [
    "Австралия",
    "автомат",
    "агент",
    "адвокат",
    "Азия",
    "акт",
    "альбом",
    "Альпы",
    "Америка",
    "амфибия",
    "ангел",
    "Англия",
    "Антарктида",
    "аппарат",
    "Атлантида",
    "Африка",
    "ацтек",
    "бабочка",
    "база",
    "Байкал",
    "банк",
    "баня",
    "бар",
    "барьер",
    "бассейн",
    "батарея",
    "башня",
    "берёза",
    "Берлин",
    "Бермуды",
    "билет",
    "биржа",
    "блин",
    "блок",
    "боевик",
    "бокс",
    "болезнь",
    "больница",
    "бомба",
    "боров",
    "борт",
    "ботинок",
    "бочка",
    "брак",
    "бревно",
    "бумага",
    "бутылка",
    "бык",
    "вагон",
    "вал",
    "ведьма",
    "век",
    "венец",
    "вертолёт",
    "верфь",
    "вес",
    "ветер",
    "взгляд",
    "вид",
    "вилка",
    "вирус",
    "вода",
    "водолаз",
    "вождь",
    "воздух",
    "война",
    "волна",
    "вор",
    "время",
    "высота",
    "газ",
    "галоп",
    "гвоздь",
    "гений",
    "Германия",
    "гигант",
    "глаз",
    "Голливуд",
    "голова",
    "горло",
    "горн",
    "гранат",
    "гребень",
    "Греция",
    "гриф",
    "груша",
    "дама",
    "декрет",
    "день",
    "десна",
    "динозавр",
    "диск",
    "доктор",
    "дракон",
    "дробь",
    "дума",
    "дух",
    "дыра",
    "дятел",
    "Европа",
    "Египет",
    "единорог",
    "ёрш",
    "жизнь",
    "жила",
    "жук",
    "журавль",
    "залог",
    "замок",
    "заноза",
    "запад",
    "запах",
    "заяц",
    "звезда",
    "зебра",
    "земля",
    "знак",
    "золото",
    "зона",
    "зуб",
    "игла",
    "игра",
    "икра",
    "Индия",
    "институт",
    "кабинет",
    "кавалер",
    "кадр",
    "казино",
    "камень",
    "камера",
    "канал",
    "караул",
    "карлик",
    "карта",
    "каша",
    "кенгуру",
    "кентавр",
    "кетчуп",
    "киви",
    "кисть",
    "кит",
    "Китай",
    "клетка",
    "ключ",
    "кокетка",
    "кол",
    "колода",
    "колонна",
    "кольцо",
    "команда",
    "конёк",
    "контрабандист",
    "концерт",
    "кора",
    "корабль",
    "королева",
    "король",
    "корона",
    "коса",
    "кость",
    "косяк",
    "кошка",
    "край",
    "кран",
    "крест",
    "кролик",
    "крошка",
    "круг",
    "крыло",
    "кулак",
    "курс",
    "лад",
    "лазер",
    "лама",
    "ласка",
    "лев",
    "лёд",
    "лейка",
    "лес",
    "лимузин",
    "линия",
    "липа",
    "лист",
    "лицо",
    "ложе",
    "Лондон",
    "лошадь",
    "лук",
    "луна",
    "луч",
    "масло",
    "масса",
    "мат",
    "машина",
    "мёд",
    "медведь",
    "Мексика",
    "мелочь",
    "место",
    "механизм",
    "микроскоп",
    "миллионер",
    "мир",
    "морковь",
    "мороженое",
    "Москва",
    "мост",
    "мотив",
    "мушка",
    "мышь",
    "налёт",
    "наряд",
    "небоскрёб",
    "ниндзя",
    "нож",
    "номер",
    "норка",
    "нота",
    "ночь",
    "НьюЙорк",
    "няня",
    "область",
    "облом",
    "образ",
    "образование",
    "обрез",
    "овсянка",
    "огонь",
    "Олимп",
    "опера",
    "операция",
    "орган",
    "орёл",
    "осьминог",
    "отель",
    "падение",
    "палата",
    "палец",
    "палочка",
    "панель",
    "пара",
    "парашют",
    "парк",
    "партия",
    "пассаж",
    "паук",
    "пачка",
    "Пекин",
    "перевод",
    "перемена",
    "перо",
    "перчатка",
    "пилот",
    "пингвин",
    "пирамида",
    "пират",
    "пистолет",
    "плата",
    "платье",
    "площадь",
    "пляж",
    "побег",
    "повар",
    "подкова",
    "подъём",
    "покров",
    "пол",
    "поле",
    "полис",
    "полиция",
    "помёт",
    "порода",
    "посольство",
    "поток",
    "почка",
    "пояс",
    "право",
    "предложение",
    "предприниматель",
    "прибор",
    "привод",
    "призрак",
    "принцесса",
    "пришелец",
    "пробка",
    "проводник",
    "проказа",
    "прокат",
    "проспект",
    "профиль",
    "путь",
    "Пушкин",
    "развод",
    "разворот",
    "рак",
    "раковина",
    "раствор",
    "рейд",
    "Рим",
    "робот",
    "рог",
    "род",
    "рок",
    "рубашка",
    "рукав",
    "рулетка",
    "рыба",
    "рысь",
    "рыцарь",
    "салют",
    "сантехник",
    "Сатурн",
    "свет",
    "свидетель",
    "секрет",
    "секция",
    "сердце",
    "сеть",
    "сила",
    "скат",
    "смерть",
    "снаряд",
    "снег",
    "снеговик",
    "собака",
    "совет",
    "солдат",
    "соль",
    "состав",
    "спутник",
    "среда",
    "ссылка",
    "стадион",
    "стан",
    "станок",
    "ствол",
    "стекло",
    "стена",
    "стойка",
    "стол",
    "стопа",
    "стрела",
    "строй",
    "струна",
    "стул",
    "ступень",
    "судьба",
    "супергерой",
    "такса",
    "танец",
    "тарелка",
    "театр",
    "телескоп",
    "течение",
    "титан",
    "Токио",
    "точка",
    "трава",
    "треугольник",
    "труба",
    "туба",
    "тур",
    "ударник",
    "удел",
    "узел",
    "урал",
    "урна",
    "утка",
    "утконос",
    "учёный",
    "учитель",
    "факел",
    "фаланга",
    "фига",
    "флейта",
    "фокус",
    "форма",
    "Франция",
    "хвост",
    "хлопок",
    "центр",
    "церковь",
    "частица",
    "червь",
    "шар",
    "шоколад",
    "шпагат",
    "шпион",
    "штат",
    "шуба",
    "экран",
    "эльф",
    "эфир",
    "Юпитер",
    "яблоко",
    "яд",
    "язык",
    "якорь",
    "ясли"
  ]
}
