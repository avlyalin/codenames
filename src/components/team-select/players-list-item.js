import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Badge } from 'src/components/badge';
import { ListGroupItem } from 'src/components/list-group';
import { TEAMS } from 'src/data/constants';

function PlayersListItem({ captainId, currentUserId, user, ...other }) {
  const isCaptain = user.id === captainId;
  const isCurrentUser = user.id === currentUserId;
  let badgeColor = 'default';
  if (user.team === TEAMS['blue']) {
    badgeColor = 'blue';
  } else if (user.team === TEAMS['red']) {
    badgeColor = 'red';
  }

  return (
    <ListGroupItem
      classes={classnames('flex justify-between', {
        'font-bold': isCurrentUser,
      })}
      {...other}
    >
      {user.name}
      {isCaptain && (
        <Badge color={badgeColor}>
          <FontAwesomeIcon icon="crown" size={'sm'} /> Капитан
        </Badge>
      )}
    </ListGroupItem>
  );
}

PlayersListItem.propTypes = {
  captainId: PropTypes.string.isRequired,
  currentUserId: PropTypes.string.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    team: PropTypes.string.isRequired,
  }).isRequired,
};

export { PlayersListItem };
