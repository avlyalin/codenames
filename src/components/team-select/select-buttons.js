import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Button } from 'src/components/button';
import { TEAMS } from 'src/data/constants';
import { Icon } from 'src/components/icon';

function SelectButtons({ userTeam, onClickJoin, ...other }) {
  const isBlueTeam = userTeam === TEAMS['blue'];
  const isRedTeam = userTeam === TEAMS['red'];
  const handleClick = (team) => {
    if (team !== userTeam) onClickJoin(team);
  };
  let iconColor = 'default';
  if (isBlueTeam) {
    iconColor = TEAMS['blue'];
  } else if (isRedTeam) {
    iconColor = TEAMS['red'];
  }

  return (
    <div className="relative flex flex-no-wrap shadow-lg" {...other}>
      <Button
        color={isBlueTeam ? TEAMS['blue'] : 'default'}
        fullWidth={true}
        rounded={false}
        shadow={false}
        onClick={() => handleClick(TEAMS['blue'])}
      >
        {!isBlueTeam && <FontAwesomeIcon icon={'user-plus'} size={'sm'} />}{' '}
        Синие
      </Button>

      <Icon
        classes={'absolute left-0 right-0 mx-auto self-center'}
        color={iconColor}
        icon="users"
      />

      <Button
        color={isRedTeam ? TEAMS['red'] : 'default'}
        fullWidth={true}
        rounded={false}
        shadow={false}
        onClick={() => handleClick(TEAMS['red'])}
      >
        {!isRedTeam && <FontAwesomeIcon icon={'user-plus'} size={'sm'} />}{' '}
        Красные
      </Button>
    </div>
  );
}

SelectButtons.propTypes = {
  userTeam: PropTypes.oneOf([TEAMS['blue'], TEAMS['red'], '']),
  onClickJoin: PropTypes.func.isRequired,
};

export { SelectButtons };
