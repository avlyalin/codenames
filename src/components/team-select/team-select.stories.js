import React from 'react';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import { containerDecorator } from '_storybook/container';
import { TEAMS } from 'src/data/constants';
import { TeamSelect } from './team-select';

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Components/Team select',
  decorators: [containerDecorator(), withKnobs],
};

export const WithUsers = () => {
  const redCaptainId = text('Red captain id', '1');
  const blueCaptainId = text('Blue captain id', '1');
  const name = text('Name', 'Blue');
  const user = {
    id: '1',
    name: text('User name', 'Killer 3000'),
    team: select('User team', TEAMS, TEAMS['blue']),
  };
  const users = [
    user,
    { name: 'player_2', team: TEAMS['blue'], id: '2' },
    { name: 'player_3', team: TEAMS['blue'], id: '3' },
    { name: 'player_4', team: TEAMS['blue'], id: '4' },
    { name: 'player_5', team: TEAMS['blue'], id: '5' },
    { name: 'player_6', team: TEAMS['red'], id: '6' },
  ];
  const captains = {
    [TEAMS['blue']]: blueCaptainId,
    [TEAMS['red']]: redCaptainId,
  };
  return (
    <TeamSelect
      name={name}
      captains={captains}
      currentUser={user}
      users={users}
      onJoin={() => {}}
      onJoinAsCaptain={() => {}}
    />
  );
};
