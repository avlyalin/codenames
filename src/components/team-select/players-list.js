import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup } from 'src/components/list-group';
import { PlayersListItem } from 'src/components/team-select/players-list-item';

function PlayersList({ captainId, currentUser, users }) {
  const playersList = users.reduce((accumulator, user) => {
    if (user.team === currentUser.team) {
      const player = (
        <PlayersListItem
          key={user.id}
          captainId={captainId}
          currentUserId={currentUser.id}
          user={user}
        />
      );
      accumulator.push(player);
    }
    return accumulator;
  }, []);
  return <ListGroup rounded={false}>{playersList}</ListGroup>;
}

PlayersList.propTypes = {
  captainId: PropTypes.string.isRequired,
  currentUser: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    team: PropTypes.string.isRequired,
  }).isRequired,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      team: PropTypes.string.isRequired,
    }).isRequired,
  ),
};

export { PlayersList };
