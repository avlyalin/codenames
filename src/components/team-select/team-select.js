import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'src/components/button';
import { SelectButtons } from 'src/components/team-select/select-buttons';
import { PlayersList } from 'src/components/team-select/players-list';
import { TEAMS } from 'src/data/constants';

const TeamSelect = React.forwardRef(function TeamSelect(
  { currentUser, captains, users, onJoin, onJoinAsCaptain, ...other },
  ref,
) {
  let color = 'default';
  let captainId = '';
  if (currentUser.team === TEAMS['blue']) {
    color = 'blue';
    captainId = captains[TEAMS['blue']];
  } else if (currentUser.team === TEAMS['red']) {
    color = 'red';
    captainId = captains[TEAMS['red']];
  }
  const handleClick = () => {
    if (currentUser.id !== captainId) onJoinAsCaptain(currentUser.team);
  };

  return (
    <div
      ref={ref}
      className="w-full border-t border-b border-solid border-gray-200 rounded-lg overflow-hidden"
      {...other}
    >
      <SelectButtons userTeam={currentUser.team} onClickJoin={onJoin} />
      <PlayersList
        captainId={captainId}
        currentUser={currentUser}
        users={users}
      />
      <Button
        color={color}
        rounded={false}
        size={'md'}
        fullWidth={true}
        shadow={false}
        onClick={handleClick}
      >
        Стать капитаном
      </Button>
    </div>
  );
});

TeamSelect.propTypes = {
  captains: PropTypes.shape({
    blue: PropTypes.string.isRequired,
    red: PropTypes.string.isRequired,
  }).isRequired,
  currentUser: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    team: PropTypes.string.isRequired,
  }).isRequired,
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      team: PropTypes.string.isRequired,
    }).isRequired,
  ),
  onJoin: PropTypes.func.isRequired,
  onJoinAsCaptain: PropTypes.func.isRequired,
};

export { TeamSelect };
