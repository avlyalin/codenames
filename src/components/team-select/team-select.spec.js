import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { TeamSelect } from 'src/components/team-select/team-select';
import { TEAMS } from 'src/data/constants';

let currentUser;
let users;
let captains;

beforeEach(() => {
  currentUser = {
    id: '1',
    name: 'player_1',
    team: TEAMS['blue'],
  };
  users = [
    currentUser,
    { id: '2', name: 'player_2', team: TEAMS['red'] },
    { id: '3', name: 'player_3', team: TEAMS['blue'] },
    { id: '4', name: 'player_4', team: TEAMS['red'] },
  ];
  captains = {
    [TEAMS['blue']]: '1',
    [TEAMS['red']]: '4',
  };
});

function setup(props) {
  const finalProps = {
    currentUser,
    users,
    captains,
    onJoin: jest.fn(),
    onJoinAsCaptain: jest.fn(),
    ...props,
  };
  const { container, ...utils } = render(<TeamSelect {...finalProps} />);
  const element = container.firstChild;
  return { element, ...utils };
}

describe('<TeamSelect />', () => {
  test('renders correctly structured component', () => {
    const { element } = setup();
    expect(element).toMatchSnapshot();
  });

  test("should render user and user's team list", () => {
    const { rerender, queryByText } = setup();

    expect(queryByText(currentUser.name)).toBeTruthy();
    expect(queryByText('player_3')).toBeTruthy();
    expect(queryByText('player_2')).toBeFalsy();
    expect(queryByText('player_4')).toBeFalsy();

    currentUser.team = TEAMS['red'];
    rerender(
      <TeamSelect
        captains={captains}
        currentUser={currentUser}
        users={users}
        onJoin={() => {}}
        onJoinAsCaptain={() => {}}
      />,
    );

    expect(queryByText(currentUser.name)).toBeTruthy();
    expect(queryByText('player_3')).toBeFalsy();
    expect(queryByText('player_2')).toBeTruthy();
    expect(queryByText('player_4')).toBeTruthy();
  });
  describe('prop: onJoin', () => {
    test('should fire event on click join another team button', () => {
      const onJoin = jest.fn();
      const { getByText } = setup({ onJoin });

      const joinButton = getByText('Красные');
      fireEvent.click(joinButton);

      expect(onJoin).toHaveBeenCalledTimes(1);
      expect(onJoin).toHaveBeenCalledWith(TEAMS['red']);
    });
    test('should not fire event on click join current team', () => {
      const onJoin = jest.fn();
      const { getByText } = setup({ onJoin });

      const joinButton = getByText('Синие');
      fireEvent.click(joinButton);

      expect(onJoin).not.toHaveBeenCalled();
    });
  });

  describe('prop: onJoinAsCaptain', () => {
    test('should fire event if user is not the captain', () => {
      const onJoinAsCaptain = jest.fn();
      captains['blue'] = '';
      const { getByText } = setup({ onJoinAsCaptain });

      const joinButton = getByText('Стать капитаном');
      fireEvent.click(joinButton);

      expect(onJoinAsCaptain).toHaveBeenCalledTimes(1);
      expect(onJoinAsCaptain).toHaveBeenCalledWith(TEAMS['blue']);
    });
    test('should not fire event if user is the captain', () => {
      const onJoinAsCaptain = jest.fn();
      const { getByText } = setup({ onJoinAsCaptain });

      const joinButton = getByText('Стать капитаном');
      fireEvent.click(joinButton);

      expect(onJoinAsCaptain).not.toHaveBeenCalled();
    });
  });
  test("should render captain's badge for the captain", () => {
    const { rerender, getByText } = setup();

    const captain = getByText('player_1');
    const captainBadge = captain.childNodes[1];

    expect(captainBadge).toBeTruthy();
    expect(captainBadge.textContent).toEqual(expect.stringMatching(/капитан/i));

    captains['blue'] = '3';
    rerender(
      <TeamSelect
        captains={captains}
        currentUser={currentUser}
        users={users}
        onJoin={() => {}}
        onJoinAsCaptain={() => {}}
      />,
    );

    const oldCaptain = getByText('player_1');
    const newCaptain = getByText('player_3');
    const newCaptainBadge = newCaptain.childNodes[1];

    expect(oldCaptain.childNodes.length).toBe(1);
    expect(newCaptainBadge.textContent).toEqual(
      expect.stringMatching(/капитан/i),
    );
  });
});
