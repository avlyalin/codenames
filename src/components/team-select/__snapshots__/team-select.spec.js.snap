// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamSelect /> renders correctly structured component 1`] = `
<div
  class="w-full border-t border-b border-solid border-gray-200 rounded-lg overflow-hidden"
>
  <div
    class="relative flex flex-no-wrap shadow-lg"
  >
    <button
      class="align-middle text-white text-center outline-none transition-colors duration-100 ease-in-out bg-blue-100 hover:bg-blue-200 btn-md w-full"
    >
       
      Синие
    </button>
    <span
      class="inline-flex justify-center items-center rounded-full h-8 w-8 text-blue-100 bg-white absolute left-0 right-0 mx-auto self-center"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-users fa-w-20 "
        data-icon="users"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
          fill="currentColor"
        />
      </svg>
    </span>
    <button
      class="align-middle text-white text-center outline-none transition-colors duration-100 ease-in-out bg-gray-300 hover:bg-gray-400 btn-md w-full"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-user-plus fa-w-20 fa-sm "
        data-icon="user-plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          fill="currentColor"
        />
      </svg>
       
      Красные
    </button>
  </div>
  <div
    class="bg-white"
  >
    <div
      class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t flex justify-between font-bold"
    >
      player_1
      <span
        class="inline-block px-4 py-1 leading-none text-center text-sm align-middle rounded-xl text-white bg-blue-100"
        data-testid="badge"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-crown fa-w-20 fa-sm "
          data-icon="crown"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M528 448H112c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h416c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm64-320c-26.5 0-48 21.5-48 48 0 7.1 1.6 13.7 4.4 19.8L476 239.2c-15.4 9.2-35.3 4-44.2-11.6L350.3 85C361 76.2 368 63 368 48c0-26.5-21.5-48-48-48s-48 21.5-48 48c0 15 7 28.2 17.7 37l-81.5 142.6c-8.9 15.6-28.9 20.8-44.2 11.6l-72.3-43.4c2.7-6 4.4-12.7 4.4-19.8 0-26.5-21.5-48-48-48S0 149.5 0 176s21.5 48 48 48c2.6 0 5.2-.4 7.7-.8L128 416h384l72.3-192.8c2.5.4 5.1.8 7.7.8 26.5 0 48-21.5 48-48s-21.5-48-48-48z"
            fill="currentColor"
          />
        </svg>
         Капитан
      </span>
    </div>
    <div
      class="py-2 px-6 border border-solid border-gray-200 border-t-0 first:border-t flex justify-between"
    >
      player_3
    </div>
  </div>
  <button
    class="align-middle text-white text-center outline-none transition-colors duration-100 ease-in-out bg-blue-100 hover:bg-blue-200 btn-md w-full"
  >
    Стать капитаном
  </button>
</div>
`;
